### PR TITLE
(WHO-1325) Add batch mirror creation ajax helper

### PR DESCRIPTION
**Pull request by thrnio**
This adds a new ajax helper, `CreateRepoMirrorsBatch`, which supports creating
multiple local repository mirrors from a single ajax call. It will create new
remote repos if they don't exist, or use the existing ones if they do.

### CODE REVIEW

File: src/main/java/com/distelli/europa/models/ContainerRepo.java [hunk 1]
```diff
@@ -71,4 +74,16 @@ public String getPullCommand()
...
+
+    /**
+     * Check if a repository name is valid.
+     *
+     * A repository name is considered valid if it matches the regular
+     * expression {@code [a-zA-Z0-9_.-]+}.
+     * @param repoName the repository name to check
+     * @return true if the name is valid, false if it is not
+     */
+    public static boolean isValidName(String repoName) {
```
Comment (brimworks) [inline src/main/java/com/distelli/europa/models/ContainerRepo.java]: I would suggest moving this into a non-static method within ContainerDb. If you move it there, then you can also more easily check if that name is already being used.

File: src/main/java/com/distelli/europa/ajax/CreateRepoMirrorsBatch.java [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+            .name(request.getSourceRepoName())
+            .provider(cred.getProvider())
+            .region(cred.getRegion())
+            .local(false)
+            .publicRepo(false)
+            .mirror(false)
+            .credId(cred.getId())
+            .id(CompactUUID.randomUUID().toString())
+            .overviewId(CompactUUID.randomUUID().toString())
+            .syncDestinationContainerRepoIds(new HashSet<>(Collections.singletonList(destinationRepoId)))
```
Comment (brimworks) [inline src/main/java/com/distelli/europa/ajax/CreateRepoMirrorsBatch.java]: Note that you can just use `Collections.singleton(destinationRepoId)`:

https://docs.oracle.com/javase/8/docs/api/java/util/Collections.html#singleton-T-

**Comment by thrnio** [comment]
@brimworks okay, I think I addressed the feedback

**Comment by brimworks** [comment]
...specifically, I think the code would be much simpler and easier to read if this method:

```
public Map<String, RepoNameValidity> validateLocalNames(String domain, Collection<String> repoN) {
}
```

...was simply:

```
public RepoNameValidity validateLocalName(String domain, String repoName) {
}
```

I can see no benefit in "batching" this.

**Comment by thrnio** [comment]
@brimworks yeah, I thought it would be simpler to do it in the method, but since I needed to differentiate the multiple cases anyway, and my data wasn't just a list of strings, it wasn't. I made it singular, which definitely reduces the complexity of the method and has a pretty minor benefit to the code using it.

**Pull request merged**